src/mon/OSDMonitor.cc

```diff
@@ -2976,6 +2982,12 @@ bool OSDMonitor::prepare_boot(MonOpRequestRef op)
     pending_inc.new_hb_front_up[from] = m->hb_front_addrs;
 
     down_pending_out.erase(from);  // if any
+    int type = osdmap.crush->get_type_id(g_conf().get_val<string>("mon_custom_subtree_down_out_type"));
+    int parent_id = osdmap.crush->get_parent_of_type(from, type);
+    auto i = down_pending_out_subtree.find(parent_id);
+    if (i != down_pending_out_subtree.end()) {
+      down_pending_out_subtree.erase(i);
+    }
```

Comment:
Can't we just directly invoke the ```erase``` call instead of doing a find-and-erase?

Reply:
yeah, we could. thanks